.github/workflows/build.yml

```diff
@@ -53,16 +52,28 @@ jobs:
 
       - name: Determine composer cache directory on Linux
         if: matrix.os == 'ubuntu-latest'
-        run: echo "::set-env name=COMPOSER_CACHE_DIR::$(composer config cache-dir)"
+        id: composer-cache-ubuntu
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
 
       - name: Determine composer cache directory on Windows
         if: matrix.os == 'windows-latest'
-        run: ECHO "::set-env name=COMPOSER_CACHE_DIR::~\AppData\Local\Composer"
+        id: composer-cache-windows
+        run: echo "::set-output name=dir::~\AppData\Local\Composer"
 
-      - name: Cache dependencies installed with composer
+      - name: Cache dependencies installed with composer on Linux
+        if: matrix.os == 'ubuntu-latest'
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache-ubuntu.outputs.dir }}
+          key: php${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-composer-
+
+      - name: Cache dependencies installed with composer on Windows
```

Comment:
We can use one step for linux and windows. See https://github.com/yiisoft/package-template/blob/d3a99793d83219481edae91553ce8d1a5e93de0d/.github/workflows/build.yml#L68

Reply:
fix